src/store/rewards/saga.ts

```diff
@@ -68,19 +68,30 @@ export function* syncRewardsAndTokenPrice() {
   if (!featureFlags.enableRewards) {
     return;
   }
-
-  yield spawn(syncFetchRewards);
   yield spawn(syncMEOWPrice);
+  yield spawn(syncFetchRewards);
+}
+
+export function* closeRewardsTooltipAfterDelay() {
+  yield delay(3000);
+  yield call(closeRewardsTooltip);
 }
 
 export function* checkNewRewardsLoaded() {
   const { meowPreviousDay, meow } = yield select((state) => state.rewards);
   const isFirstTimeLogin = yield select((state) => getDeepProperty(state, 'registration.isFirstTimeLogin'));
+  const meowTokenPrice = yield select((state) => state.rewards.meowInUSD);
+  if (meowTokenPrice === 0) {
+    yield call(fetchCurrentMeowPriceInUSD);
+  }
 
   if (!isFirstTimeLogin && meowPreviousDay !== '0') {
```

Comment:
Can we move the `isFirstTimeLogin` check higher up so we don't actually have to do any of the other stuff if it's your first time logging in?

Reply:
yes